### PR TITLE
Add Dev Container Configuration Files

### PR DESCRIPTION
**Pull request by benz0li**
For use with [GitHub Codespaces](https://docs.github.com/en/codespaces/overview) and VS Code [Dev Containers](https://code.visualstudio.com/docs/devcontainers/containers).

### CODE REVIEW

File: lectures/lecture01.ipynb [hunk 1]
```diff
@@ -28,6 +28,16 @@
...
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "08d63bd7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Pkg.activate(Base.current_project())"
```
Comment (bkamins) [inline lectures/lecture01.ipynb:39]: why is this needed? Also - does it work (i.e. without using `using Pkg` first?

**Comment by bkamins** [comment]
Could you also please add a section to README.md with the explanation of what you added and how users can use it?

File: lectures/lecture01.ipynb (hunk 1 above)
Comment (benz0li) [reply lectures/lecture01.ipynb:39]: > does it work (i.e. without using `using Pkg` first?

With the Dev Containers: Yes. Because of https://github.com/b-data/julia-docker-stack/blob/e0b072105d6b12c0722b5f104320fee7daba1f22/base/conf/julia/etc/julia/startup.jl#L8.

But not in general. I will add `using Pkg`.

> why is this needed?

To make absolutely sure, it works across different setups and configurations.

In Jupyter Notebooks, project environments are not automatically activated in every case.

Comment (bkamins) [reply lectures/lecture01.ipynb:39]: >  I will add using Pkg.

Thanks

> In Jupyter Notebooks, project environments are not automatically activated in every case.

Indeed this is the case.

**Comment by benz0li** [comment]
> Could you also please add a section to README.md with the explanation of what you added and how users can use it?

I will do so for Codespaces. If you want to this:

1. Open https://github.com/benz0li/JuliaForDataAnalysis/tree/add-dev-containers
2. Click the **`<> Code`** button, then click the **Codespaces** tab.  
   A message is displayed at the bottom of the dialog telling you who will pay for the codespace.
2. Create your codespace, either using the default options, or after configuring advanced options:
    * **Use the default options.**  
      To create a codespace using the default options, click `+`.  
      codespaces

– [Creating a codespace for a repository - GitHub Docs](https://docs.github.com/en/codespaces/developing-in-codespaces/creating-a-codespace-for-a-repository#creating-a-codespace-for-a-repository)

VS Code + Dev Containers requires either Docker or Podman. We came to the conclusion that

> this[^1] quickly gets quite complicated if the user does not know how to use Docker and needs to be given all setup instructions for different platforms [...]

[^1]: Docker

**Comment by bkamins** [comment]
> I will do so for Codespaces

I would recommend something simpler like:

"folder such and such contains configuration files for this and that. if you want to use this functionality please follow instructions given at some link"

In this way we will not have to update the README.md file in case something changes on GitHub side.

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Comment by benz0li** [comment]
ℹ️ The Dev Containers for this project are derived from https://github.com/b-data/data-science-devcontainers.
